smartsim/_core/utils/serialize.py

```diff
@@ -0,0 +1,221 @@
+            )
+            for db, nodes_info in manifest.databases
+        ],
+        "ensemble": [
+            _dictify_ensemble(
+                ens, member_info, telemetry_data_root / "ensemble"
+            )
+            for ens, member_info in manifest.ensembles
+        ],
+    }
+    try:
+        with open(manifest_file, "r", encoding="utf-8") as file:
+            manifest_dict = json.load(file)
+    except (FileNotFoundError, json.JSONDecodeError):
+        manifest_dict = {
+            "experiment": {
+                "name": manifest.metadata.exp_name,
+                "path": manifest.metadata.exp_path,
+                "launcher": manifest.metadata.launcher_name,
+            },
+            "runs": [new_run],
+        }
+    else:
+        manifest_dict["runs"].append(new_run)
+    finally:
+        with open(manifest_file, "w", encoding="utf-8") as file:
+            json.dump(manifest_dict, file, indent=2)
+
+
+def _dictify_model(
+    model: Model,
+    step_id: t.Optional[str],
+    task_id: t.Optional[str],
+    managed: t.Optional[bool],
+    out_file: str,
+    err_file: str,
+    telemetry_data_path: Path,
+) -> t.Dict[str, t.Any]:
+    colo_settings = (model.run_settings.colocated_db_settings or {}).copy()
+    db_scripts = t.cast("t.List[DBScript]", colo_settings.pop("db_scripts", []))
+    db_models = t.cast("t.List[DBModel]", colo_settings.pop("db_models", []))
+    return {
+        "name": model.name,
+        "path": model.path,
+        "run_settings": _dictify_run_settings(model.run_settings),
+        "batch_settings": _dictify_batch_settings(model.batch_settings)
+        if model.batch_settings
+        else None,
+        "params": model.params,
+        "files": {
+            "Symlink": model.files.link,
+            "Configure": model.files.tagged,
+            "Copy": model.files.copy,
+        }
+        if model.files
+        else {
+            "Symlink": [],
+            "Configure": [],
+            "Copy": [],
+        },
+        "colocated_db": {
+            "settings": colo_settings,
+            "scripts": [
+                {
+                    script.name: {
+                        "backend": "TORCH",
+                        "device": script.device,
+                    }
+                }
+                for script in db_scripts
+            ],
+            "models": [
+                {
+                    model.name: {
+                        "backend": model.backend,
+                        "device": model.device,
+                    }
+                }
+                for model in db_models
+            ],
+        }
+        if colo_settings
+        else None,
+        "telemetry_metadata": {
+            "status_dir": str(telemetry_data_path / model.name),
+            "step_id": step_id,
+            "task_id": task_id,
+            "managed": managed,
+        },
+        "out_file": out_file,
+        "error_file": err_file,
```

Comment:
Can we rename this to `err_file` for consistency?

Reply:
Whoops! Consider it renamed!